runtime/src/test/java/com/redhat/ipaas/runtime/ConnectorsITCase.java

```diff
@@ -55,4 +66,39 @@ public void connectorsGetTest() {
         assertThat(connector.getId()).contains("twitter");
     }
 
+    @Test
+    public void verifyGoodTwitterConnectionSettings() throws IOException {
+        Properties credentials = new Properties();
+        try (InputStream is = getClass().getResourceAsStream("/valid-twitter-keys.properties")) {
+            credentials.load(is);
+        }
+
+        ResponseEntity<Verifier.Result> response = post("/api/v1/connectors/twitter/verifier/connectivity", credentials, Verifier.Result.class);
```

Comment:
@redhat-ipaas/backend  more as a general remark: Actual its not a good idea to design integration tests (which actually are mis-used as unit tests here btw) to call out to external systems which we do not have under control. Since we operate Keycloak on our own, its ok to test the real thing here, too. But verifying a real Twitter connection from the various CI systems we are using will be always flaky (credentials might run out, network outage, firewall rules, .....) 

Eg. currently for the merge build I get now a 

```
Failed tests: 
  ConnectorsITCase.verifyBadTwitterConnectionSettings:101 
Expecting actual not to be empty
  ConnectorsITCase.verifyGoodTwitterConnectionSettings:80 expected:<[OK]> but was:<[ERROR]>
```

What I'm supposed to do now ? Fix the connector (but its works) ? Fix the connection from our Jenkins to Twitter ? Fix Twitter ?

Actually for such situation a integration test mock framework like [Citrus](http://www.citrusframework.org/) is very useful here to make the tests more robust, but still checking whether the app works against an (supposed to be current) external API. I think thats the best compromise between real-world checking and robustness. Of cause there should be an additional layer of _real_ integration tests, but with these one should accept that failure is a regular and somewhat expected outcome. It is important then to check whether the failure is a temporary, technical problem or more fundamentally, whether the API has changed (in which case the twitter-mocks in the other tests need to be adapted).